dotnet/src/Connectors/Connectors.AssemblyAI/AudioToText/AssemblyAIAudioToTextService.cs

```diff
@@ -0,0 +1,292 @@
+        );
+    }
+
+    private async Task<string> UploadFileAsync(Stream audioStream, CancellationToken ct)
+    {
+        const string URL = "https://api.assemblyai.com/v2/upload";
+        using var content = new StreamContent(audioStream);
+        content.Headers.ContentType = new MediaTypeHeaderValue("application/octet-stream");
+        using var request = new HttpRequestMessage(HttpMethod.Post, URL);
+        request.Headers.Authorization = new AuthenticationHeaderValue(this._apiKey);
+        request.Content = content;
+        using var response = await this._httpClient.SendAsync(request, ct).ConfigureAwait(false);
+        await ThrowIfNotSuccessStatusCodeAsync("Failed to upload file.", response, ct)
+            .ConfigureAwait(false);
+        var jsonStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+        var json = await JsonDocument.ParseAsync(jsonStream, cancellationToken: ct).ConfigureAwait(false);
+        return json.RootElement.GetProperty("upload_url").GetString()
+               ?? throw new AssemblyAIApiException("Property 'upload_url' expected but not found.");
+    }
+
+    private async Task<string> CreateTranscriptAsync(
+        string audioUrl,
+        PromptExecutionSettings? executionSettings = null,
+        CancellationToken ct = default
+    )
+    {
+        const string URL = "https://api.assemblyai.com/v2/transcript";
+        var jsonRequest = new JsonObject();
+        jsonRequest["audio_url"] = audioUrl;
+        if (executionSettings?.ExtensionData is not null)
+        {
+            foreach (var attribute in executionSettings.ExtensionData)
+            {
+                jsonRequest[attribute.Key] = JsonValue.Create(attribute.Value);
+            }
+        }
+
+        using var content = new StringContent(jsonRequest.ToJsonString(), Encoding.UTF8, "application/json");
+        using var request = new HttpRequestMessage(HttpMethod.Post, URL);
+        request.Headers.Authorization = new AuthenticationHeaderValue(this._apiKey);
+        request.Content = content;
+        using var response = await this._httpClient.SendAsync(request, ct).ConfigureAwait(false);
+        await ThrowIfNotSuccessStatusCodeAsync("Failed to create transcript", response, ct)
+            .ConfigureAwait(false);
+        var jsonStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+        var json = await JsonDocument.ParseAsync(jsonStream, cancellationToken: ct).ConfigureAwait(false);
+        if (json.RootElement.TryGetProperty("error", out var property))
+        {
+            throw new AssemblyAIApiException($"Failed to create transcript. Reason: {property.GetString()!}");
+        }
+
+        return json.RootElement.GetProperty("id").GetString()!;
+    }
+
+    private async Task<JsonDocument> WaitForTranscriptToProcessAsync(string transcriptId, CancellationToken ct)
+    {
+        var url = $"https://api.assemblyai.com/v2/transcript/{transcriptId}";
```

Comment:
Hardcoded url in methods is not good idea. Create at least a property BaseUri in class.

Reply:
nit: If url is used only in this method, we made a decision to keep constants closer to the place where it's used. If it's used in multiple methods, then we can place it on class level. But in general, both approaches are acceptable.